### PR TITLE
chore: add vault derivate apy

### PR DESCRIPTION
**Pull request by axejintao**

### CODE REVIEW

File: src/vaults/vaults.utils.spec.ts [hunk 1]
```diff
@@ -41,14 +45,14 @@ describe('vaults.utils', () => {
...
-        const timestamp = Number((startTime + int * 20000).toFixed());
+        const timestamp = Number((startTime + int * ONE_DAY_SECONDS * 14).toFixed());
         const block = Number((timestamp / 10000).toFixed());
         return {
           timestamp,
           harvests: [{ timestamp, block, harvested: BigNumber.from((int + 1 * 1.88e18).toString()) }],
           treeDistributions: [
-            { timestamp, block, token: TOKENS.SUSHI, amount: BigNumber.from((int + 1 * 5.77e12).toString()) },
-            { timestamp, block, token: TOKENS.FARM, amount: BigNumber.from((int + 1 * 4.42e12).toString()) },
+            { timestamp, block, token: TOKENS.BCVXCRV, amount: BigNumber.from((int + 1 * 5.77e12).toString()) },
```
Comment (axejintao) [inline src/vaults/vaults.utils.spec.ts:54]: these tokens were updated to make sure the "underlying" yield from these vaults gets added in tests

File: src/vaults/__snapshots__/vaults.utils.spec.ts.snap [hunk 2]
```diff
@@ -155,47 +155,77 @@ Array [
...
+    "maxApr": 1,
+    "minApr": 1,
+    "name": "bCVXCRV Compounding",
+    "oneDay": 1,
+    "sevenDay": 1,
+    "thirtyDay": 1,
+    "threeDay": 1,
+    "type": "bcvxcrv_compounding",
+  },
+  CachedValueSource {
```
Comment (axejintao) [inline src/vaults/__snapshots__/vaults.utils.spec.ts.snap:200]: vault emission

File: src/vaults/__snapshots__/vaults.utils.spec.ts.snap [hunk 3]
```diff
@@ -155,47 +155,77 @@ Array [
...
+    "sevenDay": 0.7464162921429416,
+    "thirtyDay": 0.7464162921429416,
+    "threeDay": 0.7464162921429416,
+    "type": "flat_bveCVX_emission",
   },
   CachedValueSource {
     "address": "0x19D97D8fA813EE2f51aD4B4e04EA08bAf4DFfC28",
-    "addressValueSourceType": "0x19D97D8fA813EE2f51aD4B4e04EA08bAf4DFfC28_flat_FARM_emission",
-    "apr": 28.571981817603227,
+    "addressValueSourceType": "0x19D97D8fA813EE2f51aD4B4e04EA08bAf4DFfC28_bvecvx_compounding",
```
Comment (axejintao) [inline src/vaults/__snapshots__/vaults.utils.spec.ts.snap:217]: vault additional compounding

File: src/test/tests.utils.ts [hunk 4]
```diff
@@ -115,7 +115,7 @@ export function randomSnapshots(vaultDefinition?: VaultDefinition, count?: numbe
       Object.assign(new VaultSnapshot(), {
         asset: sett.name,
         height: 0,
-        timestamp: Date.now() - 1 - i * ONE_DAY_MS,
+        timestamp: 1646105174169 - 1 - i * ONE_DAY_MS,
```
Comment (axejintao) [inline src/test/tests.utils.ts:118]: https://github.com/Badger-Finance/badger-api/issues/936

**Review by mitche50 — APPROVED** [review]

**Pull request merged**